.gitlab-ci.yml

```diff
@@ -82,11 +76,10 @@ deploy:
   extends:
     - .global_cache
     - build
-  allow_failure: false
-  # see https://gitlab.com/gitlab-org/gitlab/-/issues/28732
-  except:
-    variables:
-      - $CI_MERGE_REQUEST_ID =~ ".+"
+  rules:
+    - if: '$CI_MERGE_REQUEST_ID !~ /.+/'
+      when: always
+      allow_failure: false
   script:
```

Comment:
The regex on that issue was just straight up wrong from my testing, so I pulled the comment out to avoid confusion. You can test this pretty easily in the CI linting tool w/in Gitlab. From within a project, go to CI / CD > Pipelines, and click the 'CI Lint' button at the top. Paste this file and click the 'validate' button. Toggle back and forth off the default branch and play with the string delimiters and you'll see what I mean.